lib/web3/event.js

```diff
@@ -30,7 +30,8 @@ var watches = require('./methods/watches');
 /**
  * This prototype should be used to create event filters
  */
-var SolidityEvent = function (json, address) {
+var SolidityEvent = function (web3, json, address) {
```

Comment:
Events are just higher layer of abstraction of the logs (and that's why they are using filters internally). I find the name fine. Also if we decide to change it, it should not be a part of this pr.